### PR TITLE
.NET client update

### PR DESCRIPTION
**Pull request by stefanb995**
Adding two new parameters to Managed Instance (ProxyOverride and PublicDataEndpointEnabled)
Link to the PR with swagger change on which this code is generated: [link](https://github.com/Azure/azure-rest-api-specs/pull/5138)

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by dsgouda** [comment]
@jaredmoo please approve before I post my review.

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Discussed offline - needs enum for proxy property.

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Looks good so far, please update at least 1 scenario test to use the new properties

**Comment by stefanb995** [comment]
Updated existing scenario test adding these two new parameters.

**Review by jaredmoo — APPROVED** [review]

File: src/SDKs/SqlManagement/Management.Sql/Generated/SqlManagementClient.cs [hunk 1]
```diff
@@ -565,33 +552,6 @@ public SqlManagementClient(ServiceClientCredentials credentials, params Delegati
...
-        /// </param>
-        /// <param name='httpClient'>
-        /// HttpClient to be used
-        /// </param>
-        /// <param name='disposeHttpClient'>
-        /// True: will dispose the provided httpClient on calling SqlManagementClient.Dispose(). False: will not dispose provided httpClient</param>
-        /// <exception cref="System.ArgumentNullException">
-        /// Thrown when a required parameter is null
-        /// </exception>
-        public SqlManagementClient(ServiceClientCredentials credentials, HttpClient httpClient, bool disposeHttpClient) : this(httpClient, disposeHttpClient)
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Management.Sql/Generated/SqlManagementClient.cs]: This is a breaking change, please bump version to `1.28.0-preview`

File: src/SDKs/SDKs/_metadata/sql_resource-manager.txt [hunk 2]
```diff
@@ -3,12 +3,12 @@ AutoRest installed successfully.
 Commencing code generation
 Generating CSharp code
 Executing AutoRest command
-cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=E:\git\azure-sdk-for-net\src\SDKs
```
Comment (dsgouda) [inline src/SDKs/SDKs/_metadata/sql_resource-manager.txt]: There should be only 1 .txt file modified. Please checkout both .txt files and regenerate the code when the spec is merged.

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/SqlManagement/Management.Sql/Generated/SqlManagementClient.cs (hunk 1 above)
Comment (jaredmoo) [reply src/SDKs/SqlManagement/Management.Sql/Generated/SqlManagementClient.cs]: Deepak, is this changed expected from latest autorest? Just want to check.

Comment (jaredmoo) [reply src/SDKs/SqlManagement/Management.Sql/Generated/SqlManagementClient.cs]: Stefan, when you update csproj with new version, please also wipe out old release notes from 1.27.0 and add your new release notes.

Comment (dsgouda) [reply src/SDKs/SqlManagement/Management.Sql/Generated/SqlManagementClient.cs]: I'm not a 100% sure, but this is coming from AutoRest. @stefanb995 could you install latest autorest version and regen the code (`autorest --reset`)

**Comment by stefanb995** [comment]
Hey guys, I have regenerated code from master branch. 
autorest version is 2.0.4283. 
There are some new files generated. Can you please take a look and tell me is this ok.

**Comment by jaredmoo** [comment]
Please update csproj with new version (1.28.0-preview) and please also wipe out old release notes from 1.27.0 and add your new release notes.

**Comment by jaredmoo** [comment]
Looks like you picked up sensitivity labels from @bashahee 's Swagger change ( https://github.com/Azure/azure-rest-api-specs/pull/5148 ) , but we can't add that to SDK without tests. @dsgouda , what is your guidance when features get tied up like this?

**Comment by stefanb995** [comment]
Updated csproj.

Please let me know what could I do on my side. Should I just delete these new files and everything added for sensitivity labels?

**Comment by dsgouda** [comment]
> 
> 
> Looks like you picked up sensitivity labels from @bashahee 's Swagger change ( [Azure/azure-rest-api-specs#5148](https://github.com/Azure/azure-rest-api-specs/pull/5148) ) , but we can't add that to SDK without tests. @dsgouda , what is your guidance when features get tied up like this?

@shahabhijeet could you please take a look

**Comment by dsgouda** [comment]
@jaredmoo If I understand correctly, you are not ready to release the SesitivityLabels feature yet but you would like to release the changes in this PR.
The recommended way to do this is to remove the sensitivityLabel features from the default tag in readme.md so that generated code does not include that spec. Let us know if you have any doubts here

**Comment by bashahee** [comment]
@jaredmoo @dsgouda Can we please merge these clients? I updated readme.md and swagger files in order to generate clients (our GA is at end of the month), but I see that @stefanb995 already did that, so can our clients be merged along with current PR?

**Comment by jaredmoo** [comment]
@bashahee we need e2e scenario tests for all APIs, Can you please implement those?

**Comment by bashahee** [comment]
> @bashahee we need e2e scenario tests for all APIs, Can you please implement those?

Yes, I will implement E2E scenario tests once this PR is merged. Is that what you meant?

**Comment by jaredmoo** [comment]
I mean can you make your own PR that has your own scenario tests? We can merge them both together. (I can help resolve merge during Redmond hours if needed)

**Comment by stefanb995** [comment]
Guys, I need this merged as soon as possible. I need this change to be published to nuget before I send PR for powershell update.

**Comment by dsgouda** [comment]
@stefanb995 @jaredmoo please ensure that the code was regenerated after @bashahee 's changes so that this PR has exactly what you plan to publish in the SDK?
Also please fix the failing tests, please check if the tests are running locally in the Playback mode

**Comment by dsgouda** [comment]
@stefanb995 also link the Rest spec PR for which this SDK is being generated

**Comment by stefanb995** [comment]
@dsgouda I generated this SDK on master branch of Rest spec. I have also attached PR for my Rest spec change.

I forgot to commit session record for the test I updated. I have updated it now.  When I run tests in playback mode all tests are passing and 5 tests are being skipped.

**Comment by dsgouda** [comment]
@azuresdkci Test azure-sdk-for-net.client please

**Comment by dsgouda** [comment]
@azuresdkci retest this please

**Review by dsgouda — APPROVED** [review]

**Comment by jaredmoo** [comment]
We need to make sure that @bashahee  has scenario tests for sensitivity labels before publishing

**Comment by stefanb995** [comment]
Can @bashahee update AssemblyInfo.cs file with AssemblyFileVersion (1.28.0.0)?

**Comment by bashahee** [comment]
@stefanb995 @jaredmoo 
I opened a PR containing scenario tests for sensitivity labels clients.
https://github.com/Azure/azure-sdk-for-net/pull/5240

Moreover, I updated AssemblyInfo.cs.

**Pull request merged**